lib/chef/resource/apt_package.rb

```diff
@@ -52,6 +52,21 @@ class AptPackage < Chef::Resource::Package
         options '--no-install-recommends'
       end
       ```
+
+      **Prevent the apt_package resource from installing packages with pattern matching names**:
+
+      By default, the apt_package resource will install the named package.
+      If it can't find a package with the exact same name, it will treat the package name as regex string and match with any package that matches that regex.
+      This may lead Chef Infra Client to install one or more packages with names that match that regex.
+
+      In this example, `anchor_package_regex true` prevents the apt_package resource from installing matching packages if it can't find the `lua5.3` package.
```

Comment:
```suggestion
      In this example, `anchor_package_regex true` helps prevents that by adding `^` and `$` anchors around the package name. Example: `lua5.3` would be `^lua5.3$`.
```

Reply:
@IanMadd the apt package manager treats any specified package name as a regex, so we definitely need to be able to describe a regular expression in the documentation, and without the visual of the anchors, I don't think it's going to be as obvious what this setting accomplishes.

Reply:
That's insane. Things take regular expressions in Chef, how do you document that? Can we escape it? Regex.escape? There has to be some way to do that. We'll file a docs Issue here for that, lets get this solved so we can document our software.

Reply:
https://github.com/chef/chef/issues/13908

Reply:
@jaymzh @tpowell-progress 

The point isn't to remove regex from docs, it's to use plain text descriptions as much as possible. If we were documenting a tool that accepted a regular expression string, I'd be all for it. But in this case the user is entering `true` and `false`, so if we can we should avoid the regex anchors and expressions and explain this in plain text.


So what about something like this:

> In this example, `anchor_package_regex true` prevents the apt_package resource from treating `lua5.3` as a regular expression string if it can't find the `lua5.3` package. This prevents apt_package from installing similarly named packages like `lua5.3-rrd` or `liblua5.3-dev`.